File: client/src/core/types/Direction.ts
export type Direction = "nw" | "n" | "ne" | "w" | "c" | "e" | "sw" | "s" | "se";
export const directionChange = {
  nw: { dx: -1, dy: -1 },
  n: { dx: 0, dy: -1 },
  ne: { dx: 1, dy: -1 },
  w: { dx: -1, dy: 0 },
  c: { dx: 0, dy: 0 },
  e: { dx: 1, dy: 0 },
  sw: { dx: -1, dy: 1 },
  s: { dx: 0, dy: 1 },
  se: { dx: 1, dy: 1 },
};

File: common/src/commonMain/kotlin/model/map/Direction.kt
package dev.zprecreated.model.map

enum class Direction {
    NW, N, NE, W, C, E, SW, S, SE
}

val directionChange = mapOf(
    "nw" to Pair(-1, -1),
    "n"  to Pair( 0,  0),
    "ne" to Pair(-1,  1),
    "w"  to Pair(-1, -1),
    "c"  to Pair( 0,  0),
    "e"  to Pair(-1,  1),
    "sw" to Pair(-1, -1),
    "s"  to Pair( 0,  0),
    "se" to Pair(-1,  1),
)
